src/nat.rs

```diff
@@ -0,0 +1,76 @@
+//! The `nat` module assists with NAT traversal
+
+extern crate futures;
+extern crate p2p;
+extern crate tokio_core;
+
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, UdpSocket};
+
+use self::futures::Future;
+use self::p2p::UdpSocketExt;
+
+/// A data type representing a public Udp socket
+pub struct UdpSocketPair {
+    pub addr: SocketAddr, // Public address of the socket
+    pub rx: UdpSocket,    // Locally bound socket that can receive from the public address
+    pub tx: UdpSocket,    // Locally bound socket to send via public address
```

Comment:
Since "tx" is so common in blockchain for "transaction", I think we should use "sender" (and "receiver" for "rx"), which is consistent with Rust's channel naming.

Reply:
👍